generator/ServiceClientGeneratorLib/Generators/Marshallers/JsonRPCResponseUnmarshaller.tt

```diff
@@ -146,7 +159,40 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
     }
 #>
             }
+<#
+            if  (this.Config.ServiceModel.IsAwsQueryCompatible)
```

Comment:
Slight nit on the repeated code here. Thinking about if we ever have to revisit this, we might accidentally only update one location but not the other.

Not concerned about repeated code in the generated output, since that is generated. Just in the two templates we touched, would it be possible to move towards something like:

```
if (this.Config.ServiceModel.IsAwsQueryCompatible)
{
    GenerateAWSQueryCompatibleBlock();
}
else
{
#>
    return new <#=this.BaseException#>(errorResponse.Message, errorResponse.InnerException, errorResponse.Type, errorResponse.Code, errorResponse.RequestId, errorResponse.StatusCode);
<#
}

Reply:
Will add it in the `BaseResponseUnmarshaller`